sw/device/mask_rom/boot.md

```diff
@@ -209,63 +446,130 @@ crt_init() {
 
-```c
-peripheral_lockdown() {
-  // Parameters:
-  // - ROM_EXT manifest
-  // - Handles for Peripheral DIFs
+*   Read-only list of public key ids.
+*   OTP fuses to say whether a key id is revoked or not (keys start not revoked).
 
-  // **Open Q:** When and How do we lock down peripheral configuration?
-  // - We configure based on ROM_EXT manifest (signed)
-  // - Only locks down these peripherals if info is provided.
+Stored In: OTP and ROM
 
-  // What do we want to allow lockdown of?
-  // - Alert Manager
-  // - Pinmux / Padctrl
-  // - Entropy Configuration
-  // - Anything Else?
+Extensibility: None. A ROM_EXT can disable the use of a key, but cannot add new
+valid root key IDs. There will be a fixed set of root key IDs as part of a given
+Mask ROM version.
 
-  // This is explicitly a separate step from locking the key manager inputs. In
-  // particular, this depends on the ROM_EXT's choices, whereas the key manager
-  // inputs getting locked is not something the ROM_EXT can choose.
-}
-```
+### Boot Policy Structure
+
+Accessed by:
+
+*   Mask ROM (to choose ROM_EXT, and during bootstrapping to bless new ROM_EXT).
+*   ROM_EXT (during firmware update).
+
+Needs to contain:
+
+*   Identifier (so we know we're reading the right thing)
+
+    This also acts like a version number, because the Mask ROM code that parses
+    the boot policy can never be updated. Conversely, any changes to this
+    structure require new Mask ROM parsing code, which should be denoted with a
+    new identifier.
+
+*   Which ROM_EXT slot should be chosen first (2.b, 2.c.i).
+*   What to do if ROM_EXT does not validate (2.c.ii, 2.c.iii):
+    *   Try Alternate ROM_EXT; or
+    *   Fail to Boot
+*   What to do if ROM_EXT validates successfully, just before jumping to ROM_EXT:
+    *   Do nothing; or
+    *   Set current ROM_EXT as Primary if not already.
+*   Checksum (of everything else).
+
+Stored in: Flash (Info Partition)
+
+Extensibility: None. This info only controls the actions of the Mask ROM. You
+cannot add functionality to the Mask ROM of a given chip, so there's no way to
+add other information to this structure.
+
+The in-RAM representation also contains the boot reason (reset manager), because
+it needs to be checked most times the boot policy is also read.
+
+### ROM_EXT Manifest Structure
+
+Accessed by:
+
+*   Mask ROM (to validate ROM_EXT)
+*   BL0 Kernel (during firmware update).
+
+Needs to contain:
+
+*   Unsigned Area:
+    *   Identifier (so we know we're reading the right thing)
+        This also acts as a ROM_EXT manifest version (ie, the version of the
+        layout of the header).
+
+    *   Signature
+*   Signed Area:
+    *   Image Length
+    *   Entry Point
+        **Open Q:** Is this fixed or programmable? Field not needed if fixed.
+
+    *   ROM_EXT PMP Region Information
+        **Open Q:** Is this fixed or programmable? Field not needed if fixed.
+
+    *   Key Selection Area
+        *   Public Key
```

Comment:
Add bits for signing algorithm as well, in case we need to support other signature schemes.

Reply:
Do we think a single Mask ROM implementation will allow multiple different signing schemes? If not, then this is covered by the ROM_EXT manifest versioning.

Reply:
I am not sure if we will have to support other schemes, but I will try avoid adding complexity to the ROM. 

I agree with your comment on key manifest versioning.  I will follow up on this offline.